pytorch_lightning/overrides/distributed.py

```diff
@@ -58,27 +59,27 @@ def _find_tensors(obj):  # pragma: no-cover
 # In manual_optimization, we need to call reducer prepare_for_backward.
 # Note: Keep track of Pytorch DDP and update if there is a change
 # https://github.com/pytorch/pytorch/blob/v1.7.1/torch/nn/parallel/distributed.py#L626-L638
-def prepare_for_backward(model: DistributedDataParallel, output: Any):
+def prepare_for_backward(model: DistributedDataParallel, output: Any) -> None:
     # `prepare_for_backward` is `DistributedDataParallel` specific.
     if not isinstance(model, DistributedDataParallel):
         return
     if torch.is_grad_enabled() and model.require_backward_grad_sync:
-        model.require_forward_param_sync = True
+        model.require_forward_param_sync = True  # type: ignore[assignment]
```

Comment:
shall we define it as a union so allow bool?

Reply:
this is an attribute defined by PyTorch nn.Module. Given the mypy error I see, I believe they have incorrect typing for this. I don't know how I would change the type like you said.